### PR TITLE
Change dashboard layout issues #32 #40

### PR DESCRIPTION
**Pull request by SaamieVincken**
Add term-date filter issue #32 
Change dashboard layout issue #40

### CODE REVIEW

File: Epsilon.Host.Frontend/src/components/TermDateFilter.vue [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	</div>
+</template>
+
+<script lang="ts" setup>
+import { ref, onMounted, defineProps } from "vue"
+import { useStore } from "vuex"
+const store = useStore()
+
+const props = defineProps<{
+	items: Array<{ start_at: string; end_at: string }> | null
```
Comment (NealGeilen) [inline Epsilon.Host.Frontend/src/components/TermDateFilter.vue:34]: Please create an interface. 
`interface TermSelection{
start_at: string
end_at: string
}`

Then mention the time like this:

```suggestion
	items: TermSelection[] | null
```

File: Epsilon.Host.Frontend/src/components/TermDateFilter.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+const hasSetInitialItems = ref(false)
+const dateRangeMenuRef = ref<HTMLElement | null>(null)
+
+const setInitialItems = (): void => {
+	if (props.items !== null) {
+		originalItems.value = props.items
+	}
+}
+
+const filterItems = (): void => {
```
Comment (NealGeilen) [inline Epsilon.Host.Frontend/src/components/TermDateFilter.vue:55]: If I read this correctly. This function selects terms depending on two dates? 
And then the option is given to the user to select an term?

We are not looking for this. 
This date selection will need to go around the term selection. 
Essentially u want to manipulate the `currentTerm` start and end date in the vuex store

File: Epsilon.Host.Frontend/src/components/Performance/CompetenceGraph.vue [hunk 3]
```diff
@@ -76,9 +82,18 @@ const chartOptions = {
 	},
 }
 
+/* Set abbreviations for graph labels
```
Comment (NealGeilen) [inline Epsilon.Host.Frontend/src/components/Performance/CompetenceGraph.vue:85]: Remove unused code fragment

**Review by NealGeilen — CHANGES_REQUESTED** [review]

File: Epsilon.Host.Frontend/src/components/TermDateFilter.vue (hunk 2 above)
Comment (SaamieVincken) [reply Epsilon.Host.Frontend/src/components/TermDateFilter.vue:55]: Yes the function I wrote is to select two dates, and then in the SearchBox.vue component only terms within those dates will be shown. Then with the reset button the filtered dates are removed again and all terms are shown. But how I understand it now, is that you want the user to edit/update the dates of the current selected term is that correct?

File: Epsilon.Host.Frontend/src/components/TermDateFilter.vue (hunk 1 above)
Comment (SaamieVincken) [reply Epsilon.Host.Frontend/src/components/TermDateFilter.vue:34]: Should the interface be defined under Epsilon.Abstractions in the services folder? Or is there another designated place for these kinds of interfaces?

File: Epsilon.Host.Frontend/src/components/TermDateFilter.vue (hunk 2 above)
Comment (NealGeilen) [reply Epsilon.Host.Frontend/src/components/TermDateFilter.vue:55]: Correct

**Review by NealGeilen — COMMENTED** [review]
Technically, this all correct. I would like it if you also asked Eric for feedback on this because we lack Media expertise atm.

**Comment by NealGeilen** [comment]
Before asking for a final review please update the branch from develop.

**Review by NealGeilen — CHANGES_REQUESTED** [review]
Well, apparently I missed a major thing in testing. ;-)
The date range selection seems not to work.

**Comment by SaamieVincken** [comment]
Everything is working as it should now, somehow the dates came in undefined

**Review by NealGeilen — APPROVED** [review]

**Pull request merged**